analyses/compare-gistic/util/GISTIC-comparison-functions.R

```diff
@@ -14,64 +14,51 @@
   # Given the GISTIC result `amp_genes_conf_90.txt` or `del_genes_conf_90.txt`
   # files for the entire cohort or a specific histology, get the vector/data.frame
   # of genes for each amplification/deletion.
   #
   # Args:
   #  genes_file: file path to the `amp_genes.conf_90.txt` or
   #              `del_genes.conf_90.txt` file
-  #   include_peak_info: binary flag indicating weather or not to include the
-  #                      detection peak info
   #
   # Return:
   #  genes_output: a vector/data.frame with all the genes included in the file
-
-  genes_ragged_df <- data.table::fread(genes_file,
-    data.table = FALSE
-  )
-  genes_list <- as.list(genes_ragged_df)
-  genes_list <- genes_list %>%
-    # This removes any element from the list that is all NA -- most likely a
-    # result of reading in a ragged array
-    purrr::discard(~ all(is.na(.))) %>%
-    # This removes the element of the list that is essentially the "header" for
-    # the file
-    purrr::discard(~ any(str_detect(., "cytoband|q value"))) %>%
-    # Remove blanks -- result of ragged data.frame
-    purrr::modify(~ .[. != ""])
-
-  if (include_peak_info == TRUE) {
-    genes_list <- genes_list %>%
-      # Remove everything before and including the wide peak boundaries for each
-      # remaining element of the list
-      purrr::modify(~ .[-c(1:(str_which(., "chr")) - 1)])
-
-    # This will give us the data.frame of all the genes that were included
-    genes_output <-
-      data.table::rbindlist(lapply(genes_list, function(x)
-        data.frame(t(x))),
-      fill = TRUE
-      )
-
-    # Tidy the data.frame into a key and value format where the detection peak
-    # region is the key and the genes in that region are the values
+  
+  genes_ragged_list <- data.table::fread(genes_file,
+                                         data.table = FALSE)
+  
+  # Transpose data
+  genes_transposed <- genes_ragged_list %>%
+    t()
+  
+  # Make the header information in the first row the column names
+  colnames(genes_transposed) <- genes_transposed[1,]
+  
+  genes_df <- genes_transposed %>%
+    # Make into a data.frame object
+    as.data.frame() %>%
+    # Remove the row with header information
+    dplyr::filter(cytoband != "cytoband")
+  
+  # Gather the gene data
+  genes_output <- genes_df %>%
+    tidyr::gather(
+      "wide_peak",
+      "gene",-cytoband,-`q value`,-`residual q value`,-`wide peak boundaries`
+    ) %>%
+    dplyr::select(-wide_peak) %>%
+    # Remove the blanks in the `gene` column that result from the gathering step
+    dplyr::filter(gene != "")
+  
+  if (include_peak_info == FALSE) {
+    # Return only the genes information
     genes_output <- genes_output %>%
-      dplyr::rename(peak_region = X1) %>%
-      tidyr::gather("region", "gene_symbol", -peak_region) %>%
-      dplyr::select(-region) %>%
-      dplyr::filter(!(is.na(gene_symbol)))
+      dplyr::select(gene) %>%
+      dplyr::distinct()
   } else {
```

Comment:
Since you aren't changing the object name and always returning `genes_output` I think this else portion is just getting in the way. Can you drop the else part and just have `return(genes_output)`. That will make it easier to tell what is happening too.